File: 953 - Verifying an Alien Dictionary.cpp
class Solution {
public:
    bool isAlienSorted(vector<string>& words, string order) {
        int idx[26];
        for (int i = 0; i < 26; i++) {
            idx[order[i] - 'a'] = i;
        }
        for (int i = 0; i < words.size() - 1; i++) {
            string a = words[i], b = words[i + 1];
            if (a != b) {
                int n = min(a.size(), b.size());
                bool less = false;
                for (int j = 0; j < n; j++) {
                    if (a[j] != b[j]) {
                        if (idx[a[j] - 'a'] > idx[b[j] - 'a']) {
                            return false;
                        } else {
                            less = true;
                            break;
                        }
                    }
                }
                if (a.size() > b.size() && !less) {
                    return false;
                }
            }
        }
        return true;
    }
};

File: 953 - Verifying an Alien Dictionary.py
class Solution:
    def isAlienSorted(self, words: list[str], order: str) -> bool:
        idx = {c : i for i, c in enumerate(order)}

        for i in range(len(words) - 1):
            word1, word2 = words[i], words[i + 1]
            less = False
            for j in range(min(len(word1), len(word2))):
                if word1[j] != word2[j]:
                    if idx[word1[j]] > idx[word2[j]]:
                        return False
                    else:
                        less = True
                        break
            if len(word1) > len(word2) and not less:
                return False

        return True
